mlrun/execution.py

```diff
@@ -924,26 +936,50 @@ def set_if_valid(struct, key, val):
                 run_keys.inputs: {k: v.artifact_url for k, v in self._inputs.items()},
             },
             "status": {
-                "state": self._state,
                 "results": self._results,
                 "start_time": to_date_str(self._start_time),
                 "last_update": to_date_str(self._last_update),
             },
         }
 
+        # completion of runs is decided by the API
+        if self._state != "completed":
```

Comment:
what if other terminal state?

Reply:
if the execution is in error state the run should also be in error state
if the execution is aborted the run should also be aborted
as far as I see it